### ISSUE TITLE
Extend plotting packages list

### ISSUE THREAD

**Issue by joa-quim** [KronosTheLate/JuliaPackageComparisons#3]
Would it be possible to extend the comparisons of the plotting packages to the list enumerated in this [discourse post](https://discourse.julialang.org/t/comparison-of-plotting-packages/99860/2?u=joa-quim)?

**Comment by KronosTheLate** [comment]
Absolutely, that is great content! If @ChrisRackauckas is okay with it, I feel like copy-pasting that comment is a very good first approximation to what the plotting section should look like.

**Comment by ChrisRackauckas** [comment]
Sure

**Comment by KronosTheLate** [comment]
Could I ask you to copy that comment to here? I tried copy-pasting, but the markdown formatting is lost, which would take a lot of manual work to fix.

**Comment by ChrisRackauckas** [comment]
## Summary of Plotting Packages

* [Plots.jl](https://docs.juliaplots.org/) is the most used. It's probably the most documented, used in the most tutorials, and is used in many videos. 
    * Pros: Its main draw is that it has a lot of plugins to other packages through its recipes system, which means that a lot of odd things like `plot(sol::ODESolution)` or showing the sparsity of a `BandedMatrix` just works. With all of these integrations, it's normally what I would recommend first to newcomers since they will generally get the most done with the least work. It has a backend system, so you can make it generate plots via GR (the default), Plotly (i.e. make webpages), pyplot, PGFPlots (Latex output), UnicodePlots (i.e. output plots as text). This ease of use and flexibility is what its main draw is.
     * Cons: Its downside has traditionally been its startup time, though it's nearly a second now so that's fine. Its main downside now is mostly that it's not as configurable as something like Makie, and it's not as optimized if you get up to millions of points. Its flexibility means it's not just for standard plots but also for animations, building small graphical user interfaces, and building small apps. 
* [Makie](https://docs.makie.org/stable/) is probably the second most popular. It's natively Julia so it's cool in that aspect, you can see code all the way down. 
    * It's very optimized for large plots, especially with GPU acceleration via the OpenGL backend (GLMakie). It has a lot of examples these days. 
     * Its downside is that it's a bit less "first user friendly", given that its flexibility means there's a lot more options you're forced to specify everywhere. It has a recipe system now but it's fairly new and not well-integrated with most of the ecosystem, so it's not as seamless as Plots, though by 2024 I would assume that would largely be fixed. It has the longest startup time, used to be in minutes but now it's like 5-10 seconds. 
* [AlgebraOfGraphics.jl](https://aog.makie.org/stable/) is a grammar of graphics front-end to Makie. This essentially means it has an API that looks and acts like R's ggplot2. Thus it has largely the same pros and cons as Makie, since it's just calling Makie under the hood, but with the pro of being more familiar to users coming from R.
* [Gadfly](http://gadflyjl.org/stable/) is a grammar of graphics based library.
    * Pros: It's very familiar to a ggplot2 user. Its default theme is pretty nice looking.
    * Cons: It's a bit high on startup time, closer to Makie than Plots. Also, it's pretty feature poor. In particular, it is missing 3D plots, animations, the ability to make interactive apps with buttons, etc. For these reasons more and more people are adopting AlgebraOfGraphics, but if you're just doing some standard statistics it's fine.
* [Vega](https://www.queryverse.org/Vega.jl/stable) and [VegaLite](https://www.queryverse.org/VegaLite.jl/stable/) are of the same camp as Gadfly in the focus towards "standard" statistics and data science, but using wrappers to Javascript libraries. 
     * Pro: Fast startups 
     * Cons Similar to Gadfly, little to no flexibility (making apps, animations, ...) and integration with Julia libraries beyond Queryverse.
* [PlotlyLight](https://github.com/JuliaComputing/PlotlyLight.jl) is a no-frills wrapper to Plotly. 
    * Pro: No startup time
    * Cons: Requires reading the Plotly docs to know how to use it and has little flexibility or integration into Julia libraries.
* GR is a front end to a C library GR. It's actually used as the default front-end from Plots.jl. Many more people use it from Plots.jl than directly due to the integrations and docs, but it is nice for some things on its own.
    * Pros: It's fast, scales fairly well, has a fast startup time, has a nice GUI for investigating results, integrates well with ITerm, very flexible.
    * Cons: It's docs are bit difficult, and it doesn't have any integrations with Julia libraries. 
* [PGFPlotsX.jl](https://kristofferc.github.io/PGFPlotsX.jl/stable/) is a front-end to generate plots for Latex.
   * Pros: Fast startup, output to Latex which makes it easy to then further modify in publication documents.
   * Cons: Its interface is wonky, even if you are familiar with the pgfplots Latex package. This makes quite hard to use and teach. Very few integrations with Julia libraries (Measurements and Colors only?). Lacking flexibility in terms of animations and making apps, though it's quite flexible in its ability to modify the plots and make weird things.
* [UnicodePlots.jl](https://github.com/JuliaPlots/UnicodePlots.jl) is very simple, fast startup, and plots to text. Its downside of course is that text is the only output it has.
* [Gaston.jl](https://mbaz.github.io/Gaston.jl/v0.10.0/) a front-end to gnuplot.
     * Fast startup.
     * Pretty basic, lacking flexibility and integrations with Julia packages. Requires gnuplot so limitations on where it can be installed.
* [GMT.jl](https://github.com/GenericMappingTools/GMT.jl) is "generic mapping tools". It has some plotting tools [highlighted here](https://www.generic-mapping-tools.org/GMTjl_doc/examples/plotting_functions/).
    * Pros: Has good examples in the docs. Nice extra tools for maps.
    * Cons: Missing some standard plot types like trisurf, missing integrations with other Julia packages.
* [GNUPlot.jl](https://github.com/gcalderone/Gnuplot.jl) uses gnuplot under the hood.
    * Pros: Instant startup, has some interesting data science integrations for things like named datasets, very complete set of plots
    * Cons: Not the most complete documentation, requires Linux with gnuplot.

## tl;dr

Plots.jl is the most used for a reason. It's very flexible, integrates with the most Julia packages so you'll find it all throughout other docs, and it has many of the advantages of the other libraries through its backend system. Thus if you needed Latex output, use the pgfplots backend. If you needed a webpage, use the Plotly backend. Unicodeplots backend when you want text output. Or the GR default for the basics. With Julia v1.9 its startup time is much improved (and it's like sub second on v1.10 beta), which was its major complaint before. If you're going to use one plotting library and don't care too much about every little detail, then Plots.jl is a good one to go with. It's definitely not the best in any of the cases, animations are better in Makie, Latex is better in PGFPlotsX, etc., but it's capable everywhere.

Makie.jl is catching up and may be the default in the near future. It scales well and its getting all of the niceties of Plots.jl. I wouldn't learn it first if you're new to Julia (right now, though that will likely change by 2024). But if you need animations or want to add custom buttons to a window (make a quick GUI-like thing), Makie is unmatched. If it makes its standard plotting interface a bit simpler, gets a few more integrations, and thus matches Plots.jl in simplicity, it may hit a "best of most worlds" soon.

Otherwise it's a bit domain specific. If you were using Plots.jl and needed more flexibility for publication-quality plots, PGFPlotsX.jl can help. Or if you prefer grammar of graphics, AlgebraOfGraphics.jl is good. If you're a stats person you may find Gadfly or VegaLite familiar, though I wouldn't recommend them first because these don't satisfy general user needs (try making a plot of an FEM output and see what I mean).

All of these are pretty good. You have a lot of options. In the end, pick the one that suits your needs best.

**Comment by joa-quim** [comment]
@ChrisRackauckas I would like to take this opportunity that call you attention that the *cons* you referred about GMT.jl are no longer *so true*.

- [trisurf ](https://www.generic-mapping-tools.org/GMTjl_doc/documentation/modules/trisurf/index.html#trisurf)
- [Interoperability](https://www.generic-mapping-tools.org/GMTjl_doc/Interoperability/index.html#interoperability_with_some_other_packages). Yes, `DifferentialEquations`

**Comment by KronosTheLate** [comment]
I incorporated your comments in the version fo Chris' comment on the website. Thanks for the pointers.

**Comment by joa-quim** [comment]
OK, thanks. Not that I understand which are the *Missing some standard plot types* and overall there are no considerations on the plots quality and easiness of use, but let's leave it for future. I'm just not comfortable to make those considerations myself.

**Comment by KronosTheLate** [comment]
> I'm just not comfortable to make those considerations myself.

Excatly my feeling regarding the current content. A lot is written by me, but I feel like the wrong person for most of it. If you find a signle section that mentions packages you know, please concider giving it a loving hand!

**Comment by joa-quim** [comment]
I'll keep an eye on this, but my time in Julia has been spent almost only developing GMT.jl, so I don't really know much about other packages.

**Comment by KronosTheLate** [comment]
No problem, then the only thing I would appreciate was if you made sure your package is not misrepresented in the comparisons. Which is exactly what you did ^_^

**Comment by joa-quim** [comment]
Well, I still think the comparison is not fair to GMT. *Nice extra tools for maps* ???!!!. GMT is (well at least was) the most recognized CLI mapping system for ~30 years. But being one of the GMT developer the the author of GMT.jl it's not up to me to say that (hence my saying if not being comfortable make such considerations).

Anyway, I leave you with [this pointer](https://www.generic-mapping-tools.org/GMTjl_doc/comp_other_pkgs/index.html#comparison_with_other_packages) to where other package comparisons should consider GMT.jl as well.

**Comment by KronosTheLate** [comment]
I have never used GMT, and do not find plotting of maps that interesting. So to me, "Nice extra tool for maps" sounds about right (again, having never used it). But I see the conflict for you in writing about GMT.

The plotting section is in need for some serious rewriting, so perhaps the problem will fix itself. So I suggest we just see what the section looks like as it is getting ready for the real announcement of JuliaPackageComparisons, which will be when all section look alright.

**Comment by joa-quim** [comment]
> I have never used GMT, and do not find plotting of maps that interesting.

Off course. One of the beauties of this is the variety on interests between us. But with the risk of derailing the topic, mapping in itself is interesting for some, but how to represent things on the surface (and below/above) of Earth is as way larger domain. I'll point just a couple of links with some GMT examples.
- [Geophysics](https://www.generic-mapping-tools.org/GMTjl_doc/geophysics/)
- [Some tuturials](https://www.generic-mapping-tools.org/GMTjl_doc/tutorials/)
- [GMT historical gallery](https://docs.generic-mapping-tools.org/latest/gallery.html)

> So I suggest we just see what the section looks like as it is getting ready for the real announcement of JuliaPackageComparisons,

Agree.
